spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/SimpleConnection.java

```diff
@@ -69,17 +74,55 @@ public void close() {
 		}
 	}
 
+	/**
+	 * True if the connection is open.
+	 * @return true if the connection is open
+	 * @throws AutoRecoverConnectionNotCurrentlyOpenException if the connection is an
+	 * {@link AutorecoveringConnection} and is currently closed; this is required to
+	 * prevent the CCF from discarding this connection and opening a new one, in which
+	 * case the "old" connection would eventually be recovered and orphaned - also any
+	 * consumers belonging to it might be recovered too and the broker will deliver
+	 * messages to them when there is no code actually running to deal with those messages
+	 * (when using the SMLC). If we have actually closed the connection (e.g. via
+	 * CCF.resetConnection()) this will return false.
```

Comment:
I don't like reductions in public JavaDocs.
You can use `{@code CachingConnectionFactory}`.
Otherwise this is frustrating read for beginners.